### ISSUE TITLE
Throttle only on Unauthorized requests

### ISSUE THREAD

**Issue by lmexo** [nestjs/throttler#1624]
### Is there an existing issue that is already proposing this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe it

We try to throttle only requests if a certain number of unauthorized requests comes in. E.g. assume an attacker tries to find user/pw cominations. We now could throttle the whole endpoint. But we need many requests but to prevent brute force attacks we only want to apply the limit, if there appear too many requests where an authentication guard fails.

### Describe the solution you'd like

Basically the ThrottleGuard should be changed to separate functions to test test if limits are exceeded and to apply the trottle actually.

### What is the motivation / use case for changing the behavior?

see above

**Comment by jmcdo29** [comment]
Seems like you'd need to track the unauthroized attempts in a filter and have a different guard do the check for how many unauthorized attempts have happened, otherwise you'd be implementing that guard inside of the throttler guard, which seems like a bad way to lose track of the logic

**Comment by jmcdo29** [comment]
As a follow up on this, if you have some computationally heavy endpoint, and want to make sure a user doesn't call it more than twice in an hour, you would want that applied _before_ the computation is started, not after, because if it is tracked after the fact then it could be called upfront 5-10 times (or more) and create some denial of service attempt by taking the computational capacity of the server. 

Hopefully a system is made more robust than this attempt and that wouldn't happen, but if we were to change the approach of throttling to "only unauthorized requests" this could absolutely happen. If you'd like to implement this logic in your own package, that's perfectly fine, but it doesn't align with the goals of this one
